back/engines/commercial/insights/app/controllers/insights/web_api/v1/categories_controller.rb

```diff
@@ -34,27 +39,23 @@ def destroy
       end
 
       def destroy_all
-        authorize(::Insights::Category)
-        categories.destroy_all
+        view.categories.destroy_all
         processed_service.resets_flags(view)
-        status = categories.count.zero? ? :ok : 500
+        status = view.categories.count.zero? ? :ok : 500
         head status
       end
 
       private
 
-      def processed_service
-        @processed_service ||= Insights::ProcessedFlagsService.new
-      end
-
       def view
-        View.includes(:categories).find(params.require(:view_id))
+        @view ||= authorize(
+          View.includes(:categories).find(params.require(:view_id)),
```

Comment:
This is the hardest change for me to fully understand.  

My best guess: If `@view` is nil then authorise and return the `View` that includes the given categories _and_ a specific `view_id`?, and only _if_ it updates successfully?

Reply:
Haaa, I see. There are a few things going on here.

First, 
```ruby
def view
	@view ||= ...
end
```
is a pervasive idiom/pattern in ruby that performs memoization. You can see it as a lazy attribute. The attribute is only computed the first time it's used.

The second thing is that the categories controller delegates all the *permissioning* to the view policy (all categories are associated *with* a view). So here is what it's doing:
- it `find`s the right view (`params.require(:view_id)`)
- it already includes the categories along the way to avoid an extra db query when we'll access `view.categories`
- it checks that the current user is allowed to update the said view (`authorize(view, :updates)`)

Reply:
Thanks. Super-useful explanation! (saving for future reference) 🧑‍🎓